### PR TITLE
Corrected return code for Action API with invalid content

### PR DESCRIPTION
**Pull request by hsong-rh**
Changed return code to 400 (bad request) error code when posting an action with invalid content:
https://projects.engineering.redhat.com/browse/SSP-502

### CODE REVIEW

**Comment by hsong-rh** [comment]
@bzwei Please review

File: app/controllers/concerns/exception_handler.rb [hunk 1]
```diff
@@ -19,7 +19,18 @@ module ExceptionHandler
       json_response({ :message => e.message }, :forbidden)
     end
 
-    rescue_from Exceptions::UserError do |e|
+    rescue_from RBACApiClient::ApiError do |e|
```
Comment (bzwei) [inline app/controllers/concerns/exception_handler.rb]: I don't think we need to analyze any exception from an external system.

File: app/controllers/concerns/exception_handler.rb [hunk 2]
```diff
@@ -19,7 +19,18 @@ module ExceptionHandler
...
+      when 403
+        json_response({ :message => e.message }, :forbidden)
+      when 404
+        json_response({ :message => e.message }, :not_found)
+      else
+        json_response({ :message => e.message }, :bad_request)
+      end
+    end
+
+    rescue_from Exceptions::UserError, Exceptions::ApprovalError do |e|
```
Comment (bzwei) [inline app/controllers/concerns/exception_handler.rb]: Should we rescue all `ApprovalError` or only selected ones?

Comment (hsong-rh) [reply app/controllers/concerns/exception_handler.rb]: All `ApprovalError` raised so far are suitable to category into bad requests.

File: app/controllers/concerns/exception_handler.rb (hunk 1 above)
Comment (hsong-rh) [reply app/controllers/concerns/exception_handler.rb]: Ok, then we will keep to return error code 500 whenever RBAC service raises exceptions.

**Pull request merged**